apps/remote_control/lib/lexical/remote_control/search/store/backends/ets/state.ex

```diff
@@ -138,10 +145,18 @@ defmodule Lexical.RemoteControl.Search.Store.Backends.Ets.State do
   def replace_all(%__MODULE__{leader?: true} = state, entries) do
     rows = Schema.entries_to_rows(entries, current_schema())
 
-    with true <- :ets.delete_all_objects(state.table_name),
-         true <- :ets.insert(state.table_name, rows) do
-      :ok
-    end
+    {:ok, _, result} =
+      with_wal state.wal_state do
+        with true <- :ets.delete_all_objects(state.table_name),
+             true <- :ets.insert(state.table_name, rows) do
+          :ok
+        end
```

Comment:
Looking at the docs, these functions should always return `true`. I think we should be assertive here and match on true and let it crash here if that ever changes, instead of down the line because the `with` returned something unexpected. 

There may be other ets usage in the module where the same suggestion applies. 

```suggestion
        true = :ets.delete_all_objects(state.table_name)
        true = :ets.insert(state.table_name, rows)
        :ok
```